cmd/util/integrityattestation/utils.go

```diff
@@ -133,14 +134,42 @@ func ReadEnclaveAddress(attestationsPath string) (*common.Address, error) {
 	return &publicKeyAddress, nil
 }
 
-func GenerateHMAC() (hash.Hash, error) {
-	h := sha256.New
-	// TODO: In another PR, we should store this key and encrypt it using AWS KMS
-	// TODO: use some key which can be deterministic across restarts
-	hmac := hmac.New(h, []byte("test"))
+// Use HKDF to derive HMAC key from attested private key (as the pseudo random key)
+func DeriveHmac(attestationsPath string) (hash.Hash, error) {
+	awsConfig, err := awsconfig.LoadDefaultConfig(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("failed to load AWS config: %w", err)
+	}
+	kmsKeyID, err := GetAttestedKMSKeyID(awsConfig, attestationsPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get attested KMS Key ID: %w", err)
+	}
+	privateKey, err := GetAttestedPrivateKey(awsConfig, kmsKeyID, attestationsPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get attested private key: %w", err)
+	}
+
+	// underlying hash function for HKDF and HMAC
+	hash := sha256.New
+	// non-secret info value
+	info := []byte("HMAC key derivation")
+
+	// derive HMAC key using HKDF (with nil salt)
+	hkdf := hkdf.New(hash, crypto.FromECDSA(privateKey), nil, info)
```

Comment:
what does hkdf do? Why cant we just do `hmac := hmac.New(h, []byte(crypto.FromECDSA(privateKey)))`